internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,313 @@
+}
+
+// NewRegistryPod initializes the RegistryPod struct and sets defaults for empty fields
+func NewRegistryPod(kubeclient kubernetes.Interface, dbPath, bundleImage, namespace string) (*RegistryPod, error) {
+	rp := &RegistryPod{}
+
+	if rp.GRPCPort == 0 {
+		rp.GRPCPort = defaultGRPCPort
+	}
+
+	if len(strings.TrimSpace(rp.IndexImage)) < 1 {
+		rp.IndexImage = defaultIndexImage
+	}
+
+	if len(strings.TrimSpace(rp.BundleAddMode)) < 1 {
+		if rp.IndexImage == defaultIndexImage {
+			rp.BundleAddMode = SemverBundleAddMode
+		} else {
+			rp.BundleAddMode = ReplacesBundleAddMode
+		}
+	}
+
+	rp.Kubeclient = kubeclient
+	rp.DBPath = dbPath
+	rp.BundleImage = bundleImage
+	rp.Namespace = namespace
+
+	// validate the RegistryPod struct and ensure required fields are set
+	if err := rp.validate(); err != nil {
+		return nil, fmt.Errorf("error in validating registry pod struct: %v", err)
+	}
+
+	// call podForBundleRegistry() to make the pod definition
+	pod, err := rp.podForBundleRegistry()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod definition: %v", err)
+	}
+	rp.pod = pod
+
+	return rp, nil
+}
+
+// Create creates a bundle registry pod built from an index image
+// and returns error
+func (rp *RegistryPod) Create(ctx context.Context) error {
+	var pod *corev1.Pod
+
+	// Check if registry pod already exists
+	if tmpPod, err := rp.Kubeclient.CoreV1().Pods(rp.pod.Namespace).Get(ctx,
+		rp.pod.Name, metav1.GetOptions{}); err != nil {
+		// if error exists and the error is due to pod not found, then create a new pod
+		if k8serrors.IsNotFound(err) {
+			// create registry pod in kubernetes cluster
+			if pod, err = rp.Kubeclient.CoreV1().Pods(rp.pod.Namespace).Create(ctx,
+				rp.pod, metav1.CreateOptions{}); err != nil {
+				return fmt.Errorf("error creating registry pod: %v", err)
+			}
+			// assign rp.pod to the newly created pod
+			rp.pod = pod
+		} else {
+			return fmt.Errorf("error getting existing registry pod: %v", err)
+		}
+	} else {
+		// if an existing pod matching rp.pod.Name is found, assign rp.pod to the existing pod
+		rp.pod = tmpPod
+	}
+
+	return nil
+}
+
+// VerifyPodRunning calls checkPodStatus to verify pod status and returns the pod logs upon error
+func (rp *RegistryPod) VerifyPodRunning(ctx context.Context) error {
+	// upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(rp.pod.Namespace).Get(ctx,
+			rp.pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting pod %s: %w", rp.pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	if err := rp.checkPodStatus(ctx, podCheck); err != nil {
+		podLogs, logErr := rp.GetLogs(ctx)
+		if logErr != nil {
+			return fmt.Errorf("error verifying pod creation: %v: and fetching logs: %v", err, logErr)
+		}
+		if viper.GetBool(flags.VerboseOpt) &&
+			podLogs != "" {
```

Comment:
Nit: this reads cleaner
```suggestion
		if viper.GetBool(flags.VerboseOpt) && podLogs != "" {
```